seatunnel-connectors-v2/connector-neo4j/src/main/java/org/apache/seatunnel/connectors/seatunnel/neo4j/sink/Neo4jSinkWriter.java

```diff
@@ -17,52 +17,119 @@
+
 @Slf4j
 public class Neo4jSinkWriter implements SinkWriter<SeaTunnelRow, Void, Void> {
 
     private final Neo4jSinkQueryInfo neo4jSinkQueryInfo;
     private final transient Driver driver;
     private final transient Session session;
 
-    public Neo4jSinkWriter(Neo4jSinkQueryInfo neo4jSinkQueryInfo) {
+    private final SeaTunnelRowType seaTunnelRowType;
+    private final List<SeatunnelRowNeo4jValue> writeBuffer;
+    private final Integer maxBatchSize;
+
+    public Neo4jSinkWriter(
+            Neo4jSinkQueryInfo neo4jSinkQueryInfo, SeaTunnelRowType seaTunnelRowType) {
         this.neo4jSinkQueryInfo = neo4jSinkQueryInfo;
         this.driver = this.neo4jSinkQueryInfo.getDriverBuilder().build();
         this.session =
                 driver.session(
                         SessionConfig.forDatabase(
                                 neo4jSinkQueryInfo.getDriverBuilder().getDatabase()));
+        this.seaTunnelRowType = seaTunnelRowType;
+        this.maxBatchSize = Optional.ofNullable(neo4jSinkQueryInfo.getMaxBatchSize()).orElse(0);
+        this.writeBuffer = new ArrayList<>(maxBatchSize);
     }
 
     @Override
     public void write(SeaTunnelRow element) throws IOException {
+        if (neo4jSinkQueryInfo.batchMode()) {
+            writeByBatchSize(element);
+        } else {
+            writeOneByOne(element);
+        }
+    }
+
+    private void writeOneByOne(SeaTunnelRow element) {
         final Map<String, Object> queryParamPosition =
                 neo4jSinkQueryInfo.getQueryParamPosition().entrySet().stream()
                         .collect(
                                 Collectors.toMap(
                                         Map.Entry::getKey,
                                         e -> element.getField((Integer) e.getValue())));
         final Query query = new Query(neo4jSinkQueryInfo.getQuery(), queryParamPosition);
-        session.writeTransaction(
-                tx -> {
-                    tx.run(query);
-                    return null;
-                });
+        writeByQuery(query);
+    }
+
+    private void writeByBatchSize(SeaTunnelRow element) {
+        writeBuffer.add(new SeatunnelRowNeo4jValue(seaTunnelRowType, element));
+        tryWriteByBatchSize();
+    }
+
+    private void tryWriteByBatchSize() {
+        if (!writeBuffer.isEmpty() && writeBuffer.size() >= maxBatchSize) {
+            Query query = batchQuery();
+            writeByQuery(query);
+            writeBuffer.clear();
+        }
+    }
+
+    private Query batchQuery() {
+        try {
+            Value batchValues =
+                    Values.parameters(neo4jSinkQueryInfo.getBatchDataVariable(), writeBuffer);
+            return new Query(neo4jSinkQueryInfo.getQuery(), batchValues);
+        } catch (ClientException e) {
+            log.error("Failed to build cypher statement", e);
+            throw new Neo4jConnectorException(
+                    SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
```

Comment:
Why throw config validation failed? I think there could be has some problems with neo4j server